op-node/rollup/derive/ecotone_upgrade_transactions.go

```diff
@@ -0,0 +1,159 @@
+package derive
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/predeploys"
+	"github.com/ethereum-optimism/optimism/op-service/solabi"
+)
+
+const UpgradeToFuncSignature = "upgradeTo(address)"
+
+var (
+	// known address w/ zero txns
+	L1BlockDeployerAddress        = common.HexToAddress("0x4210000000000000000000000000000000000000")
+	GasPriceOracleDeployerAddress = common.HexToAddress("0x4210000000000000000000000000000000000001")
+
+	newL1BlockAddress        = crypto.CreateAddress(L1BlockDeployerAddress, 0)
+	newGasPriceOracleAddress = crypto.CreateAddress(GasPriceOracleDeployerAddress, 0)
+
+	deployL1BlockSource        = UpgradeDepositSource{Intent: "Ecotone: L1 Block Deployment"}
+	deployGasPriceOracleSource = UpgradeDepositSource{Intent: "Ecotone: Gas Price Oracle Deployment"}
+	updateL1BlockProxySource   = UpgradeDepositSource{Intent: "Ecotone: L1 Block Proxy Update"}
+	updateGasPriceOracleSource = UpgradeDepositSource{Intent: "Ecotone: Gas Price Oracle Proxy Update"}
+	enableEcotoneSource        = UpgradeDepositSource{Intent: "Ecotone: Gas Price Oracle Set Ecotone"}
+	beaconRootsSource          = UpgradeDepositSource{Intent: "Ecotone: beacon block roots contract deployment"}
+
+	enableEcotoneInput = crypto.Keccak256([]byte("setEcotone()"))[:4]
+
+	EIP4788From         = common.HexToAddress("0x0B799C86a49DEeb90402691F1041aa3AF2d3C875")
```

Comment:
I dont think we need `op-bindings/predeploys/eip4788.go` given this information

Reply:
One has the resulting contract code (still used for testing I believe), and the other has the tx deployment bytecode.